### ISSUE TITLE
GitHub Hub authentication impossible in Docker

### ISSUE THREAD

**Issue by sr258** [WeblateOrg/weblate#2552]
**Describe the bug**
If you run Weblate with in the Docker container, it is impossible to get the version control system for GitHub (with the capability of creating forks and pull requests) working without manually changing permissions inside the container. 

**To Reproduce**
1. Configure environment variables of Weblate Docker container (in particular set WEBLATE_GITHUB_USERNAME).
2. Run the container.
3. ```docker exec -it weblate_container bash```
4. Run any hub command to authenticate and obtain token.
5. Create component with GitHub VCS and try to push.

The reason for this problem is that ```~/.config/hub``` in git.py is always resolved as ```/root/.config/hub``` when you try to push the component. However, Weblate is running as user ```weblate``` and can't access this file. You theoretically could try to give weblate access to this location, but this is something I would avoid.

I've thought of a better fix for this issue [here](https://github.com/sr258/docker/commit/c8f5b3b94958e43fb73167feaf2a11274907be18#diff-06333407a0507230132851c763c1523aR18). (There are further changes in how the translate-toolkit outputs nested JSON. You can ignore them and only look at ```patches/weblate-3.4-hub-config-location.patch``` and ```start```). Would you accept this change as a pull request? (Without the JSON output changes.)

**Comment by nijel** [comment]
I think it's better to place the generated config to `/app/data/home/.config/hub`, then it should work without patching the code. Generating the config sounds like a useful addition for the docker image.

**Comment by sr258** [comment]
I agree about placing it somewhere else is better in theory because it keeps all data in one place, but the reason why I chose not put the config into ```/app/data/home``` is that this would mean the OAuth token is put into persistent storage. I wanted to avoid the token becoming known that way and completely keep into in the (transient) container.

Furthermore, won't ```~/.config/hub``` still resolve to ```/root/.config/hub``` without the patch? I've checked the value ```userconfig``` is resolved to in line 18 of git.py and it is ```/root/.config/hub```.
